web/src/filter.ts

```diff
@@ -291,7 +290,6 @@ const USER_OPERATORS = new Set([
 
 export class Filter {
     _terms: NarrowTerm[];
-    _sub?: StreamSubscription | undefined;
```

Comment:
What's the thinking around removing this?

Maybe it'll become clear after further reading, but a reason this might be useful is that its presence would confirm that the stream ID is valid and maps to a valid stream.

Reply:
It seemed like calling `stream_id()` did what was needed. I guess it doesn't confirm the stream id is valid, but it feels simpler to just work with the stream id than to have an extra variable storing the sub. I'm open to keeping it (putting it back) though!

Reply:
The extra lookups might be important in the context of performance optimization of very hot code paths, like `message_matches_search_term`, but we have an issue for optimizing that, so maybe we can deal with that later.

Reply:
Okay, I'll leave it as is for now then